src/signTx.c

```diff
@@ -362,6 +384,14 @@ static void signTx_handleInit_ui_runStep()
 		);
 	}
 
+	UI_STEP(HANDLE_INIT_STEP_SCRIPT_RUNNING_WARNING) {
+		if (ctx->numCollaterals == 0) {
+			UI_STEP_JUMP(HANDLE_INIT_STEP_CONFIRM);
+		}
+		//TODO(KoMa) what should be here exactly?
+		ui_displayPaginatedText("Warning", "Script running", this_fn);
```

Comment:
I'd go with something like "WARNING: Plutus script will be run", split into two lines so that all of the message is visible. Please try if it fits on the Ledger screen and then start a slack thread in ledger-dev to coordinate the msg with Trezor.

Reply:
Good.